.github/pull_request_template.md

```diff
@@ -0,0 +1,41 @@
+## Describe your changes
+< Summary of the changes.>
+
+< Please also include relevant motivation and context. >
+
+< List any dependencies that are required for this change. >
+
+## Type of change
+
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+
+## Checklist before requesting a review
+
+- [ ] My branch is up-to-date with the target branch - if not update your fork with the changes from the target branch (use `pull` with `--rebase` option if possible).
+- [ ] I have performed a self-review of my code
+- [ ] I have commented my code, particularly in hard-to-understand areas
+- [ ] I have added tests that prove my fix is effective or that my feature works
+- [ ] I have requested a reviewer and an assignee (assignee is responsible for merging)
+
+## Checklist for reviewers
+
+Each PR comes with its own improvements and flaws. The reviewer should check the following:
+- [ ] the code readable
+- [ ] the code well tested
+- [ ] the code documented
+- [ ] the code easy to maintain
```

Comment:
we can also leave "easy to maintain" out? The other three cover this some degree and this isn't explicit enough about how we measure this that the comment is probably superfluous

Reply:
I will remove "easy to maintain" since that is hard to measure objectively, we can always adapt this template later if we find it is missing something or doesn't quite work :)